builtin/logical/postgresql/secret_creds.go

```diff
@@ -91,12 +91,25 @@ func (b *backend) secretCredsRevoke(
 	}
 	username, ok := usernameRaw.(string)
 
+
 	// Get our connection
 	db, err := b.DB(req.Storage)
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if the role exists
+	var exists bool
+	query := fmt.Sprintf("SELECT exists (SELECT rolname FROM pg_roles WHERE rolname='%s');", username);
+	err = db.QueryRow(query).Scan(&exists)
+	if err != nil && err != sql.ErrNoRows {
+		return nil, err
+	}
+
+	if exists == false {
+		return nil, nil
+	}
```

Comment:
I actually think that this is a sane check to have. Dropping the role is the last thing that the revoke logic does, so if the role doesn't exist, there's no reason to continue.